Setup

xdamman opened the pull request:
new command `opencollective setup` to add open collective to your project (or GitHub repo)

This will add
- Backers and sponsors badge at the top of your README.md
- Mosaic of contributors and backers on your README.md
- Placeholders for sponsors on your README.md
- Invitation to donate when creating a new issue on GitHub
- CONTRIBUTING.md (or update existing one)

You can also add the `-i` option to use the interactive mode and decide what you want to add.
Or just run `opencollective setup --help` to see the list of options.

xdamman committed:
1.0.3

xdamman committed:
added commands to setup a repository

xdamman committed:
removed console.log

xdamman committed:
Added template for new issue

xdamman committed:
Added call to donate after npm install

xdamman committed:
Added backers and sponsors on the README

xdamman committed:
using read/writeJSONFile in utils

xdamman committed:
default README.md to test

xdamman committed:
make tests great again

xdamman committed:
cover the case of Copyright and License

xdamman committed:
add default CONTRIBUTING.md for tests

xdamman committed:
Only run the postinstall test on old versions of Node (0.x)

xdamman committed:
Make sure we don't load detect-indent (not compatible with old node versions) when only testing postinstall

xdamman committed:
Added CONTRIBUTING.md template

xdamman committed:
update template as a promise

xdamman committed:
Added contributors to README.md

xdamman committed:
Updated pull request message. Added CONTRIBUTING.md.

xdamman committed:
fix message, typo, small things

xdamman committed:
updated link to CONTRIBUTING.md + small fixes

xdamman committed:
fix

asood123 committed:
template fix

monkeywithacupcake committed:
Update setup.js

Added comment in front of "This pull request.." to let repo owners know that someone added them. Hopefully, this can ease concerns.

asood123 commented on src/commands/setup.js:
```diff
@@ -40,7 +40,7 @@ const fork = (org, repo, github_token) => {
 const submitPullRequest = (org, repo, projectPath, github_token) => {
   
-  let body = `This pull request adds backers and sponsors from your Open Collective https://opencollective.com/${repo} ❤️
+  let body = `Hi, I'm making updates for Open Collective. Either you or a supporter signed this repo up for open-collective. This pull request adds backers and sponsors from your Open Collective https://opencollective.com/${repo} ❤️
```
no hyphen in "open collective" :). "Open Collective" would be even better. @monkeywithacupcake

monkeywithacupcake committed:
Update setup.js

corrected the open-collective in the intro blurb

monkeywithacupcake committed:
Update README.md

Default the button for adding backers to be false on the contributors image

monkeywithacupcake committed:
Merge pull request #4 from opencollective/monkeywithacupcake-patch-1

Update README.md

monkeywithacupcake committed:
change to the newer lightweight postinstall

changing to https://github.com/opencollective/opencollective-postinstall

monkeywithacupcake committed:
Update setup-postinstall.js

remove logo

monkeywithacupcake committed:
Update setup.js

remove logo questions and code

monkeywithacupcake committed:
Update addPostInstall.js

remove references to logo

monkeywithacupcake committed:
Merge pull request #11 from opencollective/light-postinstall-patch

change to the newer lightweight postinstall

monkeywithacupcake committed:
Update README.md

add full link to graph contributor

monkeywithacupcake committed:
Update setup.js

change the npm install from opencollective to opencollective-postinstall

monkeywithacupcake committed:
Update addPostInstall.js

avoid adding opencollective and opencollective-postinstall dependency. Was adding both because opencollective-postinstall in setup.js which also calls addPostInstall.js

monkeywithacupcake committed:
Update addPostInstall.js

add quotes around `opencollective-postinstall`

monkeywithacupcake committed:
Update addPostInstall.js

correct pkg dependencies for `-` in opencollective-postinstall

monkeywithacupcake committed:
Create README.rst

Not all repos use .md. Adding the RST template.

znarf committed:
Merge pull request #14 from opencollective/monkeywithacupcake-patch-1

Create README.rst

znarf commented:
@xdamman @monkeywithacupcake Thank you for your work on this branch!

This is now an independent project! https://github.com/opencollective/opencollective-setup

Same code. All pull requests welcome!
